tests/console/commands/test_sync.py

```diff
@@ -28,3 +31,14 @@ def test_sync_option_is_passed_to_the_installer() -> None:
 def test_sync_option_not_available(tester: CommandTester) -> None:
     with pytest.raises(CleoNoSuchOptionError):
         tester.execute("--sync")
+
+
+def test_synced_installer(tester: CommandTester, mocker: MockerFixture) -> None:
```

Comment:
Can you add the same test for `self sync` in [tests/console/commands/self/test_sync.py](https://github.com/python-poetry/poetry/blob/main/tests/console/commands/self/test_sync.py) (and thereby check if the fix also works for `self sync`), please?

Reply:
It would be nice if you did. The tests are practically the same and it would allow us to avoid having to do the extra work of doing it ourselves.

Reply:
Fair enough. I extended the fix so that it also works for `self sync`.